### PR TITLE
Fixed core bug with process.env variables not being used, closes #12

### LINKED ISSUE #12: Missing .env file throws error
So it looks like my last commit #11 breaks if there is no .env file present in the root directory. I'll create a pull request soon, but not exactly sure how to structure any unit tests given the project structure... suggestions?

### PR DESCRIPTION
**Pull request by niftylettuce**

### CODE REVIEW

**Comment by niftylettuce** [comment]
tests are failing and look absolutely horrid to try to fix, here's some output:

```log
Failed tests:

  4) dotenv config reads path with encoding, parsing output to process.env:

      actual expected

      01

      expected 0 to equal 1

      at Assertion.fail (/Users/nexus/Public/dotenv/node_modules/should/cjs/should.js:205:17)
      at Assertion.value (/Users/nexus/Public/dotenv/node_modules/should/cjs/should.js:277:19)
      at /Users/nexus/Public/dotenv/test/main.js:56:34
      at Immediate.setImmediate (/Users/nexus/Public/dotenv/node_modules/lab/lib/runner.js:647:36)
      at runCallback (timers.js:781:20)
      at tryOnImmediate (timers.js:743:5)
      at processImmediate [as _immediateCallback] (timers.js:714:5)

  5) dotenv config makes load a synonym of config:

      actual expected

      01

      expected 0 to equal 1

      at Assertion.fail (/Users/nexus/Public/dotenv/node_modules/should/cjs/should.js:205:17)
      at Assertion.value (/Users/nexus/Public/dotenv/node_modules/should/cjs/should.js:277:19)
      at /Users/nexus/Public/dotenv/test/main.js:64:34
      at Immediate.setImmediate (/Users/nexus/Public/dotenv/node_modules/lab/lib/runner.js:647:36)
      at runCallback (timers.js:781:20)
      at tryOnImmediate (timers.js:743:5)
      at processImmediate [as _immediateCallback] (timers.js:714:5)

  8) dotenv config returns parsed object:

      actual expected

      {
        "test": """val"
      }

      expected Object { test: '' } to equal Object { test: 'val' } (at test, A has '' and B has 'val')

      at Assertion.fail (/Users/nexus/Public/dotenv/node_modules/should/cjs/should.js:205:17)
      at Assertion.value (/Users/nexus/Public/dotenv/node_modules/should/cjs/should.js:277:19)
      at /Users/nexus/Public/dotenv/test/main.js:90:25
      at Immediate.setImmediate (/Users/nexus/Public/dotenv/node_modules/lab/lib/runner.js:647:36)
      at runCallback (timers.js:781:20)
      at tryOnImmediate (timers.js:743:5)
      at processImmediate [as _immediateCallback] (timers.js:714:5)


3 of 22 tests failed
Test duration: 329 ms
The following leaks were detected:WebAssembly

error Command failed with exit code 1.
```

File: lib/main.js [hunk 1]
```diff
@@ -55,15 +56,20 @@ function config (options) {
     if (options.encoding) {
       encoding = options.encoding
     }
+    if (!options.assign) {
```
Comment (jcblw) [inline lib/main.js]: This overrides you default of true if someone does not pass anything.

Comment (niftylettuce) [reply lib/main.js]: @jcblw I've fixed this in my latest push <https://github.com/motdotla/dotenv/pull/226/commits/c350650b6bbc01ccf4774ccfc8a48beed748b477

**Comment by niftylettuce** [comment]
Any word here?  This is otherwise broken...

**Comment by jcblw** [comment]
@niftylettuce can you please into resolving the test first before we look into merging.

I need to go back and look into the other conversations about this same functionality. I don't think I had any objections but @maxbeatty I think had some input at the time. @motdotla any comments?

**Comment by maxbeatty** [comment]
Absolutely do not want this. It can be very dangerous to override existing variables. Please see previous comments on this topic for further explanation.

**Pull request closed without merging**